### DISCUSSION TITLE
features of add_geojson()

### DISCUSSION THREAD

**Discussion by shi093** [opengeos/leafmap#81] [Q&A] [answered]
1. When I load a layer to the map using add_geojson(), is there a way to specify whether the layer is turned on/off when the map is rendered?
2. How to customize the information box associated with each layer showing the data field values? Right now it shows all the data fields of each layer. I would like to arrange the order of the data fields displayed in the "i" box. And is there a way to turn on the information box after it is closed? Can I specify whether to display the information box when the map is rendered?
3. How to add pop up to the polygon layer? The solution provided for the point layer doesn't see to work for the polygon layer.

**Comment by giswqs** [comment, marked as answer]
1. The answer is no. See https://github.com/giswqs/leafmap/discussions/80. 
2. You can use GeoPandas to open the geojson and select a subset of the selected columns along with the geom column as a new GeoDataFrame. Then you can use the `add_gdf()` function to load the data onto the map. 
3. The answer is no. See https://github.com/jupyter-widgets/ipyleaflet/issues/837

**Reply by giswqs** [reply]
If you don't need the info box when the data layer is rendered, set `info_mode=None`. See https://leafmap.org/leafmap/#leafmap.leafmap.Map.add_geojson

> Displays the attributes by either on_hover or on_click. Any value other than "on_hover" or "on_click" will be treated as None. Defaults to "on_hover".
